velox/docs/functions/spark/math.rst

```diff
@@ -101,6 +101,20 @@ Mathematical Functions
     Returns ``x`` rounded down to the nearest integer.
     Supported types are: BIGINT and DOUBLE.
 
+.. spark:function:: hex(x) -> varchar
+
+    Converts ``x`` to hexadecimal.
+    Supported types are: BIGINT, VARBINARY and VARCHAR.
+    If the argument is a BIGINT or VARBINARY, hex returns the argument as a VARCHAR in hexadecimal format.
```

Comment:
My understanding now is that if argument is VARCHAR or VARBINARY, the results is string where each input byte is represented using 2 hex characters. If argument is a positive BIGINT, the result is a hex representation of the number (up to 16 characters). If argument is a negative BIGINT, the result is a hex representation of the ... not sure how to explain that ... 

The behavior for BIGINT input is a bit strange. Is this how it works in Spark?

Since VARCHAR and VARBINARY work the same, but BIGINT is different, it would be nice to document VARCHAR and VARBINARY together and BIGINT separately. Currently the doc combined BIGINT and VARBINARY, which is confusing.

Reply:
> The behavior for BIGINT input is a bit strange. Is this how it works in Spark?

```
spark-sql> select hex(17);
11

spark-sql> select hex(-17);
FFFFFFFFFFFFFFEF
```

`-17`'s complement is `1110 1111`